sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceivedMessage.java

```diff
@@ -102,7 +134,11 @@ public String getDeadLetterErrorDescription() {
      * @return The reason for a message that has been dead-lettered.
      */
     public String getDeadLetterReason() {
-        return deadLetterReason;
+        final Map<String, Object> properties = amqpAnnotatedMessage.getApplicationProperties();
```

Comment:
This is common, you can create a private method to aggregate these copied lines.